### DISCUSSION TITLE
Database import failed: The needed oc_appconfig line is not there which is unexpected

### DISCUSSION THREAD

**Discussion by dtretyakov** [nextcloud/all-in-one#1653] [Questions] [answered]
### Steps to reproduce
1. Make pg_dump from an existing nc installation
2. Copy it into postgres container accroding migration.md
3. Start db container

### Expected behavior
DB import was successful.

### Actual behavior

DB import fails with error message:
```
+ nc -z localhost 11000
+ GREP_STRING='Name: oc_appconfig; Type: TABLE; Schema: public; Owner:'
+ grep -q 'Name: oc_appconfig; Type: TABLE; Schema: public; Owner:' /mnt/data/database-dump.sql
+ echo 'The needed oc_appconfig line is not there which is unexpected.'
The needed oc_appconfig line is not there which is unexpected.
+ echo 'Please report this to https://github.com/nextcloud/all-in-one/issues. Thanks!'
Please report this to https://github.com/nextcloud/all-in-one/issues. Thanks!
+ exit 1
The database import failed. Please restore a backup and try again.
For further clues on what went wrong, look at the logs above.
...
```

### Host OS
Alpine

#### Nextcloud AIO version
4.0.1

#### Current channel
Stable

#### Other valuable info 
The problwm is that db import script has naive assumption that all DB tables have `oc_` prefixes, but it's not truth and in the real life they could be name like that:
```
$ grep "_appconfig" database-dump.sql
-- Name: nc_5665_appconfig; Type: TABLE; Schema: public; Owner: u86509
...
```

Consider supporting such case or reflect need to rename table prefixes to match proposed schema, e.g. `nc_5665_` -> `oc_`.

Reference: https://github.com/nextcloud/all-in-one/blob/edb616b18cb2c0a617fbc892e0837c44cb382b73/Containers/postgresql/start.sh#L84

**Comment by szaimen** [comment, marked as answer]
Hi, indeed is the best solution to edit the database dump and rename `nc_5665_` to `oc_`. Since this is very specific to instances where the tables prefix was changed, I would rather not add documentation on it. Only if this should come up regularly, I will try to do something.

**Comment by archont00** [comment]
I had tables without any prefix, so I had to add it:
```
sed 's@Name: \(.*Owner: ncadmin\)@Name: oc_\1@' -i database-dump.sql
sed 's@public\.@public\.oc_@g' -i database-dump.sql
```
A bit risky though: if there were some actual user data with string `public.`...

P.S. Later on, I found this (not tested by me):

https://help.nextcloud.com/t/change-the-dbtableprefix-on-a-running-docker-instance/81662/2
